scripts/generators/dispatch_table_helper_generator.py

```diff
@@ -30,103 +44,34 @@ def generate(self):
-            result = 'return VK_SUCCESS;' if command.returnType == 'VkResult' else result
-            result = 'return VK_FALSE;' if command.returnType == 'VkBool32' else result
-
-            out.append(f'static {prototype} {{ {result} }}\n')
-            out.extend([f'#endif // {command.protect}\n'] if command.protect else [])
-        out.append('\n')
-
-        out.append('const vvl::unordered_map<std::string, small_vector<std::string, 2, size_t>> api_extension_map {\n')
-        for command in [x for x in self.vk.commands.values() if x.version and x.device]:
-            out.append(f'    {{ "{command.name}", {{ "{command.version.name}" }} }},\n')
-        for command in [x for x in self.vk.commands.values() if x.extensions and x.device]:
-            extensions = ', '.join(f'"{x.name}"' for x in command.extensions)
-            out.append(f'    {{ "{command.name}", {{ {extensions} }} }},\n')
-        out.append('};\n')
-
         out.append('''
-// Using the above code-generated map of APINames-to-parent extension names, this function will:
-//   o  Determine if the API has an associated extension
-//   o  If it does, determine if that extension name is present in the passed-in set of device or instance enabled_ext_names
-//   If the APIname has no parent extension, OR its parent extension name is IN one of the sets, return TRUE, else FALSE
-static inline bool ApiParentExtensionEnabled(const std::string api_name, const DeviceExtensions *device_extension_info) {
-    auto has_ext = api_extension_map.find(api_name);
-    // Is this API part of an extension or feature group?
-    if (has_ext != api_extension_map.end()) {
-
-        // Was the extension for this API enabled in the CreateDevice call?
-        for (const auto& ext : has_ext->second) {
-            auto info = device_extension_info->get_info(ext.c_str());
-            if (info.state) {
-                return device_extension_info->*(info.state) == kEnabledByCreateinfo || device_extension_info->*(info.state) == kEnabledByInteraction;
-            }
-        }
-
-        // Was the extension for this API enabled in the CreateInstance call?
-        auto instance_extension_info = static_cast<const InstanceExtensions*>(device_extension_info);
-        for (const auto& ext : has_ext->second) {
-            auto inst_info = instance_extension_info->get_info(ext.c_str());
-            if (inst_info.state) {
-                return instance_extension_info->*(inst_info.state) == kEnabledByCreateinfo || device_extension_info->*(inst_info.state) == kEnabledByInteraction;
-            }
-        }
-        return false;
-    }
-    return true;
-}
-''')
-        out.append('''
-static inline void layer_init_device_dispatch_table(VkDevice device, VkLayerDispatchTable *table, PFN_vkGetDeviceProcAddr gpa) {
+static inline void layer_init_device_dispatch_table(VkDevice device, VkLayerDispatchTable *table, PFN_vkGetDeviceProcAddr gdpa) {
```

Comment:
Should the function name be prefixed with `layer_`? Couldn't this functionality be useful for the loader as well?

Reply:
In theory it could, but I have no plans on using this in the loader due to the loaders somewhat strange requirements compared to layers, especially since there are more than 2 dispatch tables.

Reply:
Gotcha 👍🏾